src/mibase.ts

```diff
@@ -16,14 +16,21 @@ class ExtendedVariable {
 	}
 }
 
-export enum RunCommand { CONTINUE, RUN, NONE }
+class VariableScope {
+	constructor (public readonly name: string, public readonly threadId: number, public readonly level: number) {
+	}
+
+	public static variableName (handle: number, name: string): string {
+		return `var_${handle}_${name}`;
```

Comment:
this changes the variable name that was previously used which was guaranteed a valid C identifier (or at least only contain everything that name could contain) to now also contain colons. Additionally this greatly increases variable name length, especially for nested variables, where I could imagine some debuggers might have some length limitations.

Everything the GDB-MI says about the parameter:
> The name parameter is the string by which the object can be referenced. It must be unique. If ‘-’ is specified, the varobj system will generate a string “varNNNNNN” automatically. It will be unique provided that one does not specify name of that format. The command fails if a duplicate name is found. 

It doesn't specify what characters are valid in there, but I would at least be cautious with other implementations with this. It would probably be better to let GDB decide on the name using `-` and storing that somewhere, though I think that would be better in a separate PR.

Do you think we should replace the : here with something identifier-safe?

Reply:
Why should this be C identifier related? There are other languages supported by GDB which don't follow the same identifier rules (including "exotics" that have hyphenated words). In general GDB can handle every name, as long as you quote it.

If the debugger can be used to get a name (and this also works with other debuggers than GDB) then this would be preferable; otherwise a "plain numeric one" seems to be better.

Reply:
hm ok it probably doesn't need to be C identifier related, but the length is definitely increasing a lot with this which I think will push forward a potential issue we had with very deep variables even earlier now.

Reply:
> this changes the variable name that was previously used which was guaranteed a valid C identifier (or at least only contain everything that name could contain) to now also contain colons. Additionally this greatly increases variable name length, especially for nested variables, where I could imagine some debuggers might have some length limitations.
> 

I don't think I changed this from the previous naming convention, so I'm not sure how this increases the name length.  Are you possibly confusing the parameter `name` here in this static function, with the `name` property in the class?  I just moved the previous functionality (from `variablesRequest`) into a function because I had to use it in `setVariableRequest` as well (for scope-level variables), as that was previously broken too.  In fact, in most cases I was able to make the variable names shorter, since due to using the handles for both the scope and variables, the scope `variablesReference` (which is the "id" encoded into the "var" name) start at a much lower number (i.e., 1000 instead of the previous `level << 16 | threadId`), which once you get a few levels deep turned into a big number.